frontend/src/host_orchestrator/orchestrator/artifacts/artifacts.go

```diff
@@ -115,31 +104,53 @@ func (h *Manager) GetBootloaderBundle(buildID, target string, fetcher Fetcher) (
 		}
 		return outDir, nil
 	}
-	return h.syncDownload(buildID+target+"bootloader", f)
+	return h.registry.Get(outDir).Run(f)
+}
+
+type downloadResult struct {
+	OutDir string
+	Error  error
+}
+
+type downloadContext struct {
+	mutex  sync.Mutex
+	result *downloadResult
+}
+
+func (c *downloadContext) Run(f func() (string, error)) (string, error) {
```

Comment:
The parameter and return value of `Run` could use `downloadResult` instead of `(string, error)`

Reply:
`downloadResult` is no longer needed.